### PR TITLE
Add video layout guide

### PR DESCRIPTION
**Pull request by xuesichao**
**Issue #:**
N/A

**Description of changes:**

Add a doc to guide builders on managing video quality in different video layouts.

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
N/A, only doc changes

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: guides/16_Video_Layout.md [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+
+### Video Lifecycle Management APIs
+
+Let us consider a scenario where a meeting has three attendees: attendee 0, attendee 1, and attendee 2. They are all sharing their local camera stream in a meeting, and attendee 2 is additionally sharing their screen via content share. The diagram below illustrates how all the video streams are received and managed from the perspective of attendee 0.
+
+![video lifecycle diagram](media://video_lifecycle.png)
+
+Typically, there are four steps involved in the lifecycle of a remote video with respect to a single receiving attendee:
+
+1. As remote attendees change their video stream, an up to date list of video sources (ignoring the receiving attendees’ own local video) is provided via the `remoteVideoSourcesDidChange` callback.
```
Comment (ldai1) [inline guides/16_Video_Layout.md:19]: nit: up to date -> up-to-date list

File: guides/16_Video_Layout.md [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+2. The receiving client will select the remote video sources (and possibly which simulcast layer) that they wish to receive, which will initiate a negotiation (or “subscription”) with the backend servers to start forwarding those videos, depending on a client’s downlink policy (explained in a later section).
```
Comment (ldai1) [inline guides/16_Video_Layout.md:20]: I don't understand the meaning in the brackets: (and possibly which simulcast layer). Do you mean which simulcast layer?

File: guides/16_Video_Layout.md (hunk 1 above)
Comment (ldai1) [inline guides/16_Video_Layout.md:19]: nit: up to date -> up-to-date list

File: guides/16_Video_Layout.md [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+
+   - Note that some downlink policies may expose APIs to allow dynamic control over this decision.
+
+3. For each subscribed or unsubscribed video, the receiving attendee will receive a `videoTileDidUpdate` or `videoTileWasRemoved` callback.
```
Comment (ldai1) [inline guides/16_Video_Layout.md:24]: Does builders need to add the `audiovideoobserver` to receive these callbacks?

File: guides/16_Video_Layout.md [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+
+**NScaleVideoUplinkBandwidthPolicy (Default):** In this policy, an attendee encodes and sends one video layer upstream and scales the resolution and bitrate based on the number of video publishers. This is to ensure the receiving attendees who are subscribed to these videos do not experience excessive network or compute load, which may degrade the meeting quality.
+
+**SimulcastUplinkPolicy:** Each attendee may encode and send up to two layers of video with different bitrates and resolutions under this policy. It enables the recipient to switch between the two video layers to adapt to changing network conditions or different layouts (e.g., a featured video may desire a higher resolution than a gallery view video).
+
+- Note: the `SimulcastUplinkPolicy` is currently only supported in Chrome 76 and above versions.
+
+#### Downlink Policy
+
+The downlink policy controls which remote video sources are received downstream based on the network condition, local resources (e.g., CPU), and other factors such as layout changes in the application. The JavaScript SDK currently provides two downlink policies:
```
Comment (ldai1) [inline guides/16_Video_Layout.md:43]: To clarify, are you trying to say: The downlink policy controls which remote video sources are received **as** downstream based on the network condition?

File: guides/16_Video_Layout.md [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+
+### Case 1: Small Grid Layout (up to 4 Videos) with Fixed Tile Size
+
+We recommend the default configuration of the `NScaleVideoUplinkBandwidthPolicy`  and  `AllHighestVideoBandwidthPolicy` when displaying a small number of video tiles of a fixed size, so the recipient is less likely to get overwhelmed. Video freezes can occur due to network congestion and recover automatically.
+
+#### JavaScript SDK Implementation
+
+Refer to code sample in [Use case 16](https://github.com/aws/amazon-chime-sdk-js#video).
+
+#### React Components Library Implementation
```
Comment (ldai1) [inline guides/16_Video_Layout.md:59]: Is this something you will move or add to the react component library documentation?

File: guides/16_Video_Layout.md [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+Let us consider a scenario where a meeting has three attendees: attendee 0, attendee 1, and attendee 2. They are all sharing their local camera stream in a meeting, and attendee 2 is additionally sharing their screen via content share. The diagram below illustrates how all the video streams are received and managed from the perspective of attendee 0.
+
+![video lifecycle diagram](media://video_lifecycle.png)
+
+Typically, there are four steps involved in the lifecycle of a remote video with respect to a single receiving attendee:
+
+1. As remote attendees change their video stream, an up-to-date list of video sources (ignoring the receiving attendees’ own local video) is provided via the `remoteVideoSourcesDidChange` callback.
+2. The receiving client will select the remote video sources (and possibly which simulcast layer) that they wish to receive, which will initiate a negotiation (or “subscription”) with the backend servers to start forwarding those videos, depending on a client’s downlink policy (explained in a later section).
+
+   - Note that some downlink policies may expose APIs to allow dynamic control over this decision.
```
Comment (devalevenkatesh) [inline guides/16_Video_Layout.md:22]: For notes, suggest using ">" syntax over "-" as dashes show bullets, and we generally use ">" for notes.

For e.g.:
```
> Note: Some downlink policies may expose APIs to allow dynamic control over this decision.
```

will render as:
> Note: Some downlink policies may expose APIs to allow dynamic control over this decision.

File: guides/16_Video_Layout.md [hunk 7]
```diff
@@ -0,0 +1,527 @@
...
+
+#### Uplink Policy
+
+The video encoding parameters are determined by the uplink policy, which takes into consideration the number of video publishers on the call. If necessary, it may suggest encoding multiple “simulcast” layers (with different bitrates and resolutions). The JavaScript SDK currently provides two uplink policies:
+
+**NScaleVideoUplinkBandwidthPolicy (Default):** In this policy, an attendee encodes and sends one video layer upstream and scales the resolution and bitrate based on the number of video publishers. This is to ensure the receiving attendees who are subscribed to these videos do not experience excessive network or compute load, which may degrade the meeting quality.
+
+**SimulcastUplinkPolicy:** Each attendee may encode and send up to two layers of video with different bitrates and resolutions under this policy. It enables the recipient to switch between the two video layers to adapt to changing network conditions or different layouts (e.g., a featured video may desire a higher resolution than a gallery view video).
+
+- Note: the `SimulcastUplinkPolicy` is currently only supported in Chrome 76 and above versions.
```
Comment (devalevenkatesh) [inline guides/16_Video_Layout.md:39]: Same as earlier comment: Use ">" instead or "-".

File: guides/16_Video_Layout.md [hunk 8]
```diff
@@ -0,0 +1,527 @@
...
+#### Downlink Policy
+
+The downlink policy controls which remote video sources are received downstream based on the network condition, local resources (e.g., CPU), and other factors such as layout changes in the application. The JavaScript SDK currently provides two downlink policies:
+
+**AllHighestVideoBandwidthPolicy (Default):** This policy subscribes to the highest quality video layer of all publishers. Under constrained network, some or all videos may freeze as packets are lost on the network, and they will recover automatically once network congestion is reduced.
+
+**VideoPriorityBasedPolicy:** This policy allows builders to choose and configure which video sources to subscribe to, their relative priorities and preferences such as, `targetDisplaySize` . Under constrained network, the policy will automatically pause video tiles based on the priority of the video sources until recovery. If used with the `SimulcastUplinkPolicy`, the policy may temporarily degrade to lower resolutions. For videos of the same priority, the policy first tries to allocate bandwidth to ensure as many videos can be displayed as possible before upgrading to higher quality if the bandwidth allows.
```
Comment (devalevenkatesh) [inline guides/16_Video_Layout.md:47]: nit: "while video sources" -> "while remote video sources"

**Pull request merged**